cmd/generator/main.go

```diff
@@ -406,55 +370,131 @@ func getFromTarFile(reader *tar.Reader, filepath string) ([]byte, error) {
 	return nil, errors.Errorf("failed to find %s in tar file", filepath)
 }
 
-func downloadSignature(asset github.ReleaseAsset) (string, error) {
-	signature, err := getSignatureFromAsset(asset)
+func downloadSignature(url string) (string, error) {
+	logger.Debugf("fetching signature file from %s", url)
+
+	resp, err := http.Get(url)
 	if err != nil {
-		return "", errors.Wrap(err, "Can't get signature from the asset")
+		return "", errors.Wrapf(err, "failed to download signature file from %s", url)
 	}
+	defer resp.Body.Close()
 
-	return signature, nil
-}
+	if resp.StatusCode != http.StatusOK {
+		return "", errors.Errorf("received %d status code while downloading plugin bundle", resp.StatusCode)
+	}
 
-func getSignatureFromAsset(asset github.ReleaseAsset) (string, error) {
-	url := asset.GetBrowserDownloadURL()
-	logger.Debugf("fetching signature file from %s", url)
+	sigFile, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to open downloaded signature file from %s", url)
+	}
+	return base64.StdEncoding.EncodeToString(sigFile), nil
+}
 
+func downloadBundleData(url string) ([]byte, error) {
 	resp, err := http.Get(url)
 	if err != nil {
-		return "", errors.Wrapf(err, "failed to download signature file %s", asset.GetName())
+		return nil, errors.Wrapf(err, "failed to download plugin bundle")
 	}
 	defer resp.Body.Close()
 
-	sigFile, err := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.Errorf("received %d status code while downloading plugin bundle", resp.StatusCode)
+	}
+
+	gzBundleReader, err := gzip.NewReader(resp.Body)
 	if err != nil {
-		return "", errors.Wrapf(err, "failed to open downloaded signature file %s", asset.GetName())
+		return nil, errors.Wrapf(err, "failed to read gzipped plugin bundle")
 	}
-	return base64.StdEncoding.EncodeToString(sigFile), nil
+
+	bundleData, err := ioutil.ReadAll(gzBundleReader)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to read plugin bundle")
+	}
+
+	return bundleData, nil
 }
 
-func getIcon(ctx context.Context, icon string) ([]byte, error) {
-	if strings.HasPrefix(icon, "http") {
-		logger.Debugf("fetching icon from url %s", icon)
+func getIconDataFromTarFile(file []byte, path string) (string, error) {
+	iconData, err := getFromTarFile(tar.NewReader(bytes.NewReader(file)), path)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to read icon data from plugin bundle for path %s", path)
+	}
 
-		req, err := http.NewRequestWithContext(ctx, http.MethodGet, icon, nil)
+	logger.Debugf("using icon specified in manifest as %s", path)
+	return fmt.Sprintf("data:image/svg+xml;base64,%s", base64.StdEncoding.EncodeToString(iconData)), nil
+}
+
+func getIconDataFromPath(ctx context.Context, path string) (string, error) {
+	var icon []byte
+	if strings.HasPrefix(path, "http") {
```

Comment:
Does this execution path matter anymore, since we have the "missing" icons always checked in?

Reply:
I was only moving this code around. I'm totally fine to remove it. Going forward I would also like to move more icons to the respective plugin repos. (See also https://community-release.mattermost.com/core/pl/4ekprbzjctrn9qpj8y53eifq5h)

Reply:
Would you be fine also removing the code to support non-svg icon, given that this is not supported by the server and the Marketplace only uses svg icons?

Reply:
Yes, let's remove the non-SVG support. We'll probably have to leave the icons checked in to keep the older plugins iconified, but even this could go eventually.